src/Hangfire.Tags.PostgreSql/TagsStorageTransactionRetryPolicy.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Generic;
+using Hangfire.Logging;
+using Hangfire.Tags.Storage;
+using Npgsql;
+
+namespace Hangfire.Tags.PostgreSql
+{
+    /// <summary>
+    /// Retry policy to address https://github.com/face-it/Hangfire.Tags/issues/35
+    /// </summary>
+    public class TagsStorageTransactionRetryPolicy : ITagsStorageTransactionRetryPolicy
+    {
+        private readonly int _retryCount;
+        private readonly ILog _logger = LogProvider.For<TagsStorageTransactionRetryPolicy>(); 
+
+        public TagsStorageTransactionRetryPolicy(int retryCount)
+        {
+            _retryCount = retryCount;
+        }
+        
+        private static readonly HashSet<string> ConcurrencyErrorCodes = new HashSet<string>
+        {
+            "40001", //serialization_failure
+            "40P01"  //deadlock_detected 
```

Comment:
`"40P01"  //deadlock_detected` is an error I discovered while running in production. It is not reproducible within the unit test, but it happens rarely.